Sources/Client/Client/Client.swift

```diff
@@ -18,6 +18,8 @@ public final class Client {
     public typealias OnTokenChange = (Token?) -> Void
     /// A WebSocket connection callback type.
     public typealias OnConnect = (Connection) -> Void
+    /// A WebSocket connected callback event type.
+    public typealias OnConnected = (Error?) -> Void
```

Comment:
The difference between `OnConnect` and `OnConnected` is not clear enough, why don't have just have 1?
```swift
public typealias OnConnect = (Connection, Error?) -> Void
// or better
public typealias OnConnect = (Result<Connection, Error>) -> Void
```

Reply:
I added it specifically for `set(user:connected:)`. 
No need for other types, this has everything:
```swift
public typealias OnConnect = (Connection) -> Void
```
But for set user and connect it's good to provide a completion callback where they can open UI when we connected. I thought to keep just onConnect, but they should handle it as `if connection == .connected {}`

If you definitly don't think it's useful, I can switch back to just `OnConnect`.

Reply:
In this case we should rename `OnConnect` to `OnConnectionChange` or `OnConnectionChanged` and keep `OnConnected` as is

Reply:
Let's just keep only `OnConnect`.